File: android/app/src/main/java/com/vyb/services/AIService.kt
/*
 * AIService - Gemini AI API integration for Android
 * Implements T050: Android AI Service Integration
 * Kotlin/Jetpack Compose counterpart to web AIService with API compatibility
 */

package com.vyb.services

import android.util.Log
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.google.gson.Gson
import com.google.gson.JsonObject
import com.google.gson.annotations.SerializedName
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import java.io.IOException
import java.net.HttpURLConnection
import java.net.URL
import java.util.*
import kotlin.random.Random

// MARK: - Error Types

sealed class AIServiceError(message: String) : Exception(message) {
    class AuthError(message: String) : AIServiceError("Authentication Error: $message")
    class ValidationError(message: String) : AIServiceError("Validation Error: $message")
    class NetworkError(message: String) : AIServiceError("Network Error: $message")
    class ApiError(message: String) : AIServiceError("API Error: $message")
    class ParseError(message: String) : AIServiceError("Parse Error: $message")
}

// MARK: - Request/Response Models

enum class AnalysisType {
    @SerializedName("trends") TRENDS,
    @SerializedName("creative") CREATIVE,
    @SerializedName("accessibility") ACCESSIBILITY,
    @SerializedName("performance") PERFORMANCE
}

enum class SuggestionType {
    @SerializedName("layout") LAYOUT,
    @SerializedName("color") COLOR,
    @SerializedName("typography") TYPOGRAPHY,
    @SerializedName("composition") COMPOSITION
}

enum class VariationType {
    @SerializedName("creative") CREATIVE,
    @SerializedName("trend-based") TREND_BASED,
    @SerializedName("accessibility") ACCESSIBILITY,
    @SerializedName("brand-aligned") BRAND_ALIGNED
}

data class CanvasAnalysisRequest(
    val canvas: DesignCanvasData,
    val deviceType: String,
    val analysisType: List<AnalysisType>,
    val userPreferences: UserPreferences?
)

data class AISuggestion(
    val id: String,
    val type: SuggestionType,
    val description: String,
    val confidence: Double,
    val preview: String?
)

data class TrendItem(
    val name: String,
    val popularity: Double,
    val description: String
)

data class TrendData(
    val category: String,
    val trends: List<TrendItem>
)

data class CanvasAnalysisResponse(
    val analysisId: String,
    val suggestions: List<AISuggestion>,
    val confidence: Double,
    val trends: TrendData?,
    val processingTime: Int
)

data class VariationRequest(
    val baseCanvas: DesignCanvasData,
    val variationType: VariationType,
    val count: Int,
    val preferences: UserPreferences?
)

data class VariationResponse(
    val requestId: String,
    val variations: List<DesignCanvasData>,
    val confidence: Double,
    val processingTime: Int
)

data class UserPreferences(
    val style: String?,
    val industry: String?,
    val targetAudience: String?,
    val brandColors: List<String>?
)

data class CurrentTrendsResponse(
    val trendsId: String,
    val categories: List<TrendData>,
    val lastUpdated: String,
    val confidence: Double
)

// MARK: - Data Transfer Objects

data class DesignCanvasData(
    val id: String,
    val deviceType: String,
    val dimensions: SimpleCanvasDimensions,
    val layers: List<LayerData>,
    val metadata: SimpleCanvasMetadata,
    val state: String
)

data class SimpleCanvasDimensions(
    val width: Double,
    val height: Double,
    val pixelDensity: Double
)

data class SimpleCanvasMetadata(
    val createdAt: Date,
    val modifiedAt: Date,
    val tags: List<String>,
    val description: String? = null,
    val author: String? = null
)

data class LayerData(
    val id: String,
    val type: String,
    val zIndex: Int,
    val content: Map<String, Any>,
    val transform: LayerTransform,
    val style: Map<String, Any>,
    val constraints: Map<String, Any>,
    val metadata: SimpleLayerMetadata
)

data class LayerTransform(
    val x: Double,
    val y: Double,
    val scaleX: Double,
    val scaleY: Double,
    val rotation: Double,
    val opacity: Double
)

data class SimpleLayerMetadata(
    val source: String,
    val createdAt: Date
)

// MARK: - AI Service Class

class AIService(
    private val apiKey: String = "",
    private val baseUrl: String = "https://ai.gemini.googleapis.com/v1"
) : ViewModel() {
    
    private val gson = Gson()
    
    var isLoading by mutableStateOf(false)
        private set
    
    var lastError by mutableStateOf<AIServiceError?>(null)
        private set
    
    companion object {
        private const val TAG = "AIService"
        private const val TIMEOUT_MS = 30000
    }
    
    // MARK: - Public API Methods
    
    /**
     * Analyze canvas for AI suggestions
     * Corresponds to POST /canvas/analyze
     */
    suspend fun analyzeCanvas(request: CanvasAnalysisRequest): CanvasAnalysisResponse {
        val startTime = System.currentTimeMillis()
        
        return withContext(Dispatchers.IO) {
            try {
                if (apiKey.isEmpty()) {
                    throw AIServiceError.AuthError("API key not configured")
                }
                
                // Validate canvas data
                validateCanvasData(request.canvas)
                
                isLoading = true
                lastError = null
                
                val url = URL("$baseUrl/canvas/analyze")
                val connection = url.openConnection() as HttpURLConnection
                
                connection.apply {
                    requestMethod = "POST"
                    setRequestProperty("Content-Type", "application/json")
                    setRequestProperty("Authorization", "Bearer $apiKey")
                    doOutput = true
                    connectTimeout = TIMEOUT_MS
                    readTimeout = TIMEOUT_MS
                }
                
                // Send request
                val requestJson = gson.toJson(request)
                connection.outputStream.use { os ->
                    os.write(requestJson.toByteArray())
                }
                
                // Handle response
                when (connection.responseCode) {
                    HttpURLConnection.HTTP_OK -> {
                        val responseJson = connection.inputStream.bufferedReader().use { it.readText() }
                        val response = gson.fromJson(responseJson, CanvasAnalysisResponse::class.java)
                        
                        val processingTime = (System.currentTimeMillis() - startTime).toInt()
                        
                        // Update processing time if not set by server
                        val finalResponse = if (response.processingTime == 0) {
                            response.copy(processingTime = processingTime)
                        } else {
                            response
                        }
                        
                        isLoading = false
                        finalResponse
                    }
                    else -> {
                        val errorMessage = try {
                            connection.errorStream?.bufferedReader()?.use { it.readText() } ?: "Unknown error"
                        } catch (e: IOException) {
                            "HTTP ${connection.responseCode}"
                        }
                        isLoading = false
                        throw AIServiceError.ApiError("HTTP ${connection.responseCode}: $errorMessage")
                    }
                }
            } catch (e: AIServiceError) {
                isLoading = false
                lastError = e
                throw e
            } catch (e: Exception) {
                val error = AIServiceError.NetworkError("Canvas analysis failed: ${e.message}")
                isLoading = false
                lastError = error
                throw error
            }
        }
    }
    
    /**
     * Generate design variations
     * Corresponds to POST /variations/generate
     */
    suspend fun generateVariations(request: VariationRequest): VariationResponse {
        val startTime = System.currentTimeMillis()
        
        return withContext(Dispatchers.IO) {
            try {
                if (apiKey.isEmpty()) {
                    throw AIServiceError.AuthError("API key not configured")
                }
                
                // Validate base canvas
                validateCanvasData(request.baseCanvas)
                
                if (request.count <= 0 || request.count > 10) {
                    throw AIServiceError.ValidationError("Count must be between 1 and 10")
                }
                
                isLoading = true
                lastError = null
                
                val url = URL("$baseUrl/variations/generate")
                val connection = url.openConnection() as HttpURLConnection
                
                connection.apply {
                    requestMethod = "POST"
                    setRequestProperty("Content-Type", "application/json")
                    setRequestProperty("Authorization", "Bearer $apiKey")
                    doOutput = true
                    connectTimeout = TIMEOUT_MS
                    readTimeout = TIMEOUT_MS
                }
                
                // Send request
                val requestJson = gson.toJson(request)
                connection.outputStream.use { os ->
                    os.write(requestJson.toByteArray())
                }
                
                // Handle response
                when (connection.responseCode) {
                    HttpURLConnection.HTTP_OK -> {
                        val responseJson = connection.inputStream.bufferedReader().use { it.readText() }
                        val response = gson.fromJson(responseJson, VariationResponse::class.java)
                        
                        val processingTime = (System.currentTimeMillis() - startTime).toInt()
                        
                        // Update processing time if not set by server
                        val finalResponse = if (response.processingTime == 0) {
                            response.copy(processingTime = processingTime)
                        } else {
                            response
                        }
                        
                        isLoading = false
                        finalResponse
                    }
                    else -> {
                        val errorMessage = try {
                            connection.errorStream?.bufferedReader()?.use { it.readText() } ?: "Unknown error"
                        } catch (e: IOException) {
                            "HTTP ${connection.responseCode}"
                        }
                        isLoading = false
                        throw AIServiceError.ApiError("HTTP ${connection.responseCode}: $errorMessage")
                    }
                }
            } catch (e: AIServiceError) {
                isLoading = false
                lastError = e
                throw e
            } catch (e: Exception) {
                val error = AIServiceError.NetworkError("Variation generation failed: ${e.message}")
                isLoading = false
                lastError = error
                throw error
            }
        }
    }
    
    /**
     * Get current design trends
     * Corresponds to GET /trends/current
     */
    suspend fun getCurrentTrends(): CurrentTrendsResponse {
        return withContext(Dispatchers.IO) {
            try {
                if (apiKey.isEmpty()) {
                    throw AIServiceError.AuthError("API key not configured")
                }
                
                isLoading = true
                lastError = null
                
                val url = URL("$baseUrl/trends/current")
                val connection = url.openConnection() as HttpURLConnection
                
                connection.apply {
                    requestMethod = "GET"
                    setRequestProperty("Authorization", "Bearer $apiKey")
                    connectTimeout = TIMEOUT_MS
                    readTimeout = TIMEOUT_MS
                }
                
                // Handle response
                when (connection.responseCode) {
                    HttpURLConnection.HTTP_OK -> {
                        val responseJson = connection.inputStream.bufferedReader().use { it.readText() }
                        val response = gson.fromJson(responseJson, CurrentTrendsResponse::class.java)
                        
                        isLoading = false
                        response
                    }
                    else -> {
                        val errorMessage = try {
                            connection.errorStream?.bufferedReader()?.use { it.readText() } ?: "Unknown error"
                        } catch (e: IOException) {
                            "HTTP ${connection.responseCode}"
                        }
                        isLoading = false
                        throw AIServiceError.ApiError("HTTP ${connection.responseCode}: $errorMessage")
                    }
                }
            } catch (e: AIServiceError) {
                isLoading = false
                lastError = e
                throw e
            } catch (e: Exception) {
                val error = AIServiceError.NetworkError("Trends request failed: ${e.message}")
                isLoading = false
                lastError = error
                throw error
            }
        }
    }
    
    // MARK: - Validation Methods
    
    private fun validateCanvasData(canvas: DesignCanvasData) {
        if (canvas.id.isEmpty()) {
            throw AIServiceError.ValidationError("Canvas ID is required")
        }
        
        if (canvas.dimensions.width <= 0 || canvas.dimensions.height <= 0) {
            throw AIServiceError.ValidationError("Canvas dimensions must be positive")
        }
        
        if (canvas.layers.isEmpty()) {
            throw AIServiceError.ValidationError("Canvas must contain at least one layer")
        }
        
        // Validate each layer
        canvas.layers.forEach { layer ->
            if (layer.id.isEmpty()) {
                throw AIServiceError.ValidationError("Layer ID is required")
            }
            
            if (layer.type.isEmpty()) {
                throw AIServiceError.ValidationError("Layer type is required")
            }
        }
    }
    
    // MARK: - Convenience Methods
    
    /**
     * Create analysis request from canvas data
     */
    fun createAnalysisRequest(
        canvasData: DesignCanvasData,
        analysisTypes: List<AnalysisType> = listOf(AnalysisType.CREATIVE, AnalysisType.TRENDS),
        userPreferences: UserPreferences? = null
    ): CanvasAnalysisRequest {
        return CanvasAnalysisRequest(
            canvas = canvasData,
            deviceType = canvasData.deviceType,
            analysisType = analysisTypes,
            userPreferences = userPreferences
        )
    }
    
    /**
     * Create variation request from canvas data
     */
    fun createVariationRequest(
        baseCanvasData: DesignCanvasData,
        variationType: VariationType = VariationType.CREATIVE,
        count: Int = 3,
        preferences: UserPreferences? = null
    ): VariationRequest {
        return VariationRequest(
            baseCanvas = baseCanvasData,
            variationType = variationType,
            count = count,
            preferences = preferences
        )
    }
    
    // MARK: - Flow-based Methods (Jetpack Compose Integration)
    
    /**
     * Analyze canvas with Flow for reactive UI
     */
    fun analyzeCanvasFlow(
        canvasData: DesignCanvasData,
        analysisTypes: List<AnalysisType> = listOf(AnalysisType.CREATIVE, AnalysisType.TRENDS),
        userPreferences: UserPreferences? = null
    ): Flow<CanvasAnalysisResponse> = flow {
        val request = createAnalysisRequest(canvasData, analysisTypes, userPreferences)
        val response = analyzeCanvas(request)
        emit(response)
    }.flowOn(Dispatchers.IO)
    
    /**
     * Generate variations with Flow for reactive UI
     */
    fun generateVariationsFlow(
        baseCanvasData: DesignCanvasData,
        variationType: VariationType = VariationType.CREATIVE,
        count: Int = 3,
        preferences: UserPreferences? = null
    ): Flow<VariationResponse> = flow {
        val request = createVariationRequest(baseCanvasData, variationType, count, preferences)
        val response = generateVariations(request)
        emit(response)
    }.flowOn(Dispatchers.IO)
    
    /**
     * Get current trends with Flow
     */
    fun getCurrentTrendsFlow(): Flow<CurrentTrendsResponse> = flow {
        val response = getCurrentTrends()
        emit(response)
    }.flowOn(Dispatchers.IO)
    
    // MARK: - Utility Methods
    
    fun clearError() {
        lastError = null
    }
    
    fun updateApiKey(newApiKey: String) {
        viewModelScope.launch {
            // In a real implementation, you might want to save this securely
            Log.d(TAG, "API key updated")
        }
    }
    
    // MARK: - Mock/Demo Methods (for development)
    
    /**
     * Generate mock analysis response for testing
     */
    fun generateMockAnalysisResponse(canvasData: DesignCanvasData): CanvasAnalysisResponse {
        val suggestions = listOf(
            AISuggestion(
                id = UUID.randomUUID().toString(),
                type = SuggestionType.COLOR,
                description = "Consider using warmer colors to improve emotional engagement",
                confidence = 0.85,
                preview = null
            ),
            AISuggestion(
                id = UUID.randomUUID().toString(),
                type = SuggestionType.LAYOUT,
                description = "Adjusting the layout hierarchy could improve visual flow",
                confidence = 0.78,
                preview = null
            )
        )
        
        val trends = TrendData(
            category = "Design Trends 2024",
            trends = listOf(
                TrendItem(
                    name = "Minimalist Design",
                    popularity = 0.92,
                    description = "Clean, simple designs with plenty of whitespace"
                ),
                TrendItem(
                    name = "Bold Typography",
                    popularity = 0.87,
                    description = "Strong, impactful font choices for better readability"
                )
            )
        )
        
        return CanvasAnalysisResponse(
            analysisId = UUID.randomUUID().toString(),
            suggestions = suggestions,
            confidence = 0.82,
            trends = trends,
            processingTime = Random.nextInt(500, 2000)
        )
    }
    
    /**
     * Generate mock variation response for testing
     */
    fun generateMockVariationResponse(
        baseCanvas: DesignCanvasData,
        count: Int = 3
    ): VariationResponse {
        val variations = (1..count).map { index ->
            baseCanvas.copy(
                id = UUID.randomUUID().toString(),
                layers = baseCanvas.layers.map { layer ->
                    layer.copy(
                        transform = layer.transform.copy(
                            opacity = (0.7 + index * 0.1).coerceAtMost(1.0)
                        )
                    )
                }
            )
        }
        
        return VariationResponse(
            requestId = UUID.randomUUID().toString(),
            variations = variations,
            confidence = 0.75,
            processingTime = Random.nextInt(800, 3000)
        )
    }
}

// MARK: - Companion Object (Singleton Pattern)

object AIServiceProvider {
    private var instance: AIService? = null
    
    fun getInstance(apiKey: String = "", baseUrl: String = "https://ai.gemini.googleapis.com/v1"): AIService {
        return instance ?: synchronized(this) {
            instance ?: AIService(apiKey, baseUrl).also { instance = it }
        }
    }
    
    fun setApiKey(apiKey: String) {
        instance?.updateApiKey(apiKey)
    }
}

File: web/src/services/AIService.ts
import { DesignCanvas } from '../entities/DesignCanvas'
import { AIServiceError } from '../types'

/**
 * Canvas analysis         try {
          baseCanvas = new DesignCanvas(request.canvas as any)
        } catch (error) {
          console.error('Canvas validation error:', error);
          throw new AIServiceError('Invalid canvas data', 'VALIDATION_ERROR')
        }st matching ai-api.yaml specification
 */
export interface CanvasAnalysisRequest {
  canvas: DesignCanvas | any
  deviceType: string
  analysisType: ('trends' | 'creative' | 'accessibility' | 'performance')[]
  userPreferences?: UserPreferences
}

/**
 * AI suggestion response structure
 */
export interface AISuggestion {
  id: string
  type: 'layout' | 'color' | 'typography' | 'composition'
  description: string
  confidence: number
  preview?: string
}

/**
 * Trend data structure
 */
export interface TrendData {
  category: string
  trends: Array<{
    name: string
    popularity: number
    description: string
  }>
}

/**
 * Canvas analysis response structure
 */
export interface CanvasAnalysisResponse {
  analysisId: string
  suggestions: AISuggestion[]
  confidence: number
  trends?: TrendData
  processingTime: number
}

/**
 * Variation generation request
 */
export interface VariationRequest {
  baseCanvas: DesignCanvas | any
  variationType: 'creative' | 'trend-based' | 'accessibility' | 'brand-aligned'
  count: number
  preferences?: UserPreferences
}

/**
 * Variation generation response
 */
export interface VariationResponse {
  requestId: string
  variations: DesignCanvas[]
  confidence: number
  processingTime: number
}

/**
 * User preferences for AI analysis
 */
export interface UserPreferences {
  style?: string
  industry?: string
  targetAudience?: string
  brandColors?: string[]
}

/**
 * Current trends response
 */
export interface CurrentTrendsResponse {
  trendsId: string
  categories: TrendData[]
  lastUpdated: string
  confidence: number
}

/**
 * AI Service for Gemini API integration
 * Implements the contract specifications from ai-api.yaml
 */
export class AIService {
  private baseUrl: string
  private apiKey: string

  constructor(apiKey?: string, baseUrl?: string) {
    // For web environment, API configuration should be passed explicitly
    this.baseUrl = baseUrl || 'https://ai.gemini.googleapis.com/v1'
    this.apiKey = apiKey || ''
  }

  /**
   * Analyze canvas for AI suggestions
   * POST /canvas/analyze
   */
  async analyzeCanvas(request: CanvasAnalysisRequest): Promise<CanvasAnalysisResponse> {
    const startTime = Date.now()
    
    try {
      if (!this.apiKey) {
        throw new AIServiceError('API key not configured', 'AUTH_ERROR')
      }

      // Validate canvas data - accept either DesignCanvas instance or plain data
      let canvas: DesignCanvas;
      if (request.canvas instanceof DesignCanvas) {
        canvas = request.canvas;
        if (!canvas.isValid()) {
          throw new AIServiceError('Invalid canvas data', 'VALIDATION_ERROR')
        }
      } else if (request.canvas && typeof request.canvas === 'object') {
        // Create DesignCanvas from plain data (e.g., from API)
        try {
          canvas = new DesignCanvas(request.canvas as any)
        } catch (error) {
          console.error('Canvas validation error:', error);
          throw new AIServiceError('Invalid canvas data', 'VALIDATION_ERROR')
        }
      } else {
        throw new AIServiceError('Canvas data is required', 'VALIDATION_ERROR')
      }

      const response = await fetch(`${this.baseUrl}/canvas/analyze`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${this.apiKey}`
        },
        body: JSON.stringify({
          canvas: canvas.toJSON(),
          deviceType: request.deviceType,
          analysisType: request.analysisType,
          userPreferences: request.userPreferences
        })
      })

      if (!response.ok) {
        const error = await response.text()
        throw new AIServiceError(`Analysis failed: ${error}`, 'API_ERROR')
      }

      const data = await response.json()
      const processingTime = Date.now() - startTime

      return {
        analysisId: data.analysisId,
        suggestions: data.suggestions,
        confidence: data.confidence,
        trends: data.trends,
        processingTime
      }
    } catch (error) {
      if (error instanceof AIServiceError) {
        throw error
      }
      throw new AIServiceError(`Canvas analysis failed: ${error}`, 'NETWORK_ERROR')
    }
  }

  /**
   * Generate design variations
   * POST /variations/generate
   */
  async generateVariations(request: VariationRequest): Promise<VariationResponse> {
    const startTime = Date.now()

    try {
      if (!this.apiKey) {
        throw new AIServiceError('API key not configured', 'AUTH_ERROR')
      }

      // Validate input - accept either DesignCanvas instance or plain data
      let baseCanvas: DesignCanvas;
      if (request.baseCanvas instanceof DesignCanvas) {
        baseCanvas = request.baseCanvas;
        if (!baseCanvas.isValid()) {
          throw new AIServiceError('Invalid base canvas', 'VALIDATION_ERROR')
        }
      } else if (request.baseCanvas && typeof request.baseCanvas === 'object') {
        // Create DesignCanvas from plain data
        try {
          baseCanvas = new DesignCanvas(request.baseCanvas as any)
        } catch (error) {
          throw new AIServiceError('Invalid base canvas', 'VALIDATION_ERROR')
        }
      } else {
        throw new AIServiceError('Base canvas is required', 'VALIDATION_ERROR')
      }

      if (request.count < 1 || request.count > 5) {
        throw new AIServiceError('Variation count must be between 1 and 5', 'VALIDATION_ERROR')
      }

      const response = await fetch(`${this.baseUrl}/variations/generate`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${this.apiKey}`
        },
        body: JSON.stringify({
          baseCanvas: baseCanvas.toJSON(),
          variationType: request.variationType,
          count: request.count,
          preferences: request.preferences
        })
      })

      if (!response.ok) {
        const error = await response.text()
        throw new AIServiceError(`Variation generation failed: ${error}`, 'API_ERROR')
      }

      const data = await response.json()
      const processingTime = Date.now() - startTime

      // Convert response variations back to DesignCanvas instances
      const variations = data.variations.map((canvasData: any) => 
        DesignCanvas.fromJSON(canvasData)
      )

      return {
        requestId: data.requestId,
        variations,
        confidence: data.confidence,
        processingTime
      }
    } catch (error) {
      if (error instanceof AIServiceError) {
        throw error
      }
      throw new AIServiceError(`Variation generation failed: ${error}`, 'NETWORK_ERROR')
    }
  }

  /**
   * Get current design trends
   * GET /trends/current
   */
  async getCurrentTrends(): Promise<CurrentTrendsResponse> {
    try {
      if (!this.apiKey) {
        throw new AIServiceError('API key not configured', 'AUTH_ERROR')
      }

      const response = await fetch(`${this.baseUrl}/trends/current`, {
        method: 'GET',
        headers: {
          'Authorization': `Bearer ${this.apiKey}`
        }
      })

      if (!response.ok) {
        const error = await response.text()
        throw new AIServiceError(`Trends request failed: ${error}`, 'API_ERROR')
      }

      const data = await response.json()

      return {
        trendsId: data.trendsId,
        categories: data.categories,
        lastUpdated: data.lastUpdated,
        confidence: data.confidence
      }
    } catch (error) {
      if (error instanceof AIServiceError) {
        throw error
      }
      throw new AIServiceError(`Trends request failed: ${error}`, 'NETWORK_ERROR')
    }
  }

  /**
   * Check if AI service is available
   */
  async healthCheck(): Promise<boolean> {
    try {
      const response = await fetch(`${this.baseUrl}/health`, {
        method: 'GET',
        headers: {
          'Authorization': `Bearer ${this.apiKey}`
        }
      })
      return response.ok
    } catch {
      return false
    }
  }

  /**
   * Configure API settings
   */
  configure(options: { apiKey?: string; baseUrl?: string }) {
    if (options.apiKey) {
      this.apiKey = options.apiKey
    }
    if (options.baseUrl) {
      this.baseUrl = options.baseUrl
    }
  }
}